miri/bin/miri.rs

```diff
@@ -64,17 +68,17 @@ impl<'a> CompilerCalls<'a> for MiriCompilerCalls {
         odir: &Option<PathBuf>,
         ofile: &Option<PathBuf>,
     ) -> Compilation {
-        self.0.late_callback(matches, sess, input, odir, ofile)
+        self.default.late_callback(matches, sess, input, odir, ofile)
     }
     fn build_controller(
         &mut self,
         sess: &Session,
         matches: &getopts::Matches,
     ) -> CompileController<'a> {
-        let mut control = self.0.build_controller(sess, matches);
+        let mut control = self.default.build_controller(sess, matches);
         control.after_hir_lowering.callback = Box::new(after_hir_lowering);
         control.after_analysis.callback = Box::new(after_analysis);
-        if std::env::var("MIRI_HOST_TARGET") != Ok("yes".to_owned()) {
+        if !self.host_target {
             // only fully compile targets on the host
```

Comment:
What do we even get from doing this on the host?

Reply:
aux builds.

Reply:
Sorry, I need more detail than that. What do aux builds have to do with anything here? We just need the mir of the dependent crates anway, no full build?

Reply:
When we do aux builds in our test suite, we need to produce an rlib that we can link against. There are no mir-only rlibs (yet). The creation of the rlib is done via platform specific tools, so that's why we need to fail here. We could move to `lld`, which should work universally.

Reply:
Ah, makes sense. Thanks :)